finat/tensor_product.py

```diff
@@ -0,0 +1,177 @@
+from __future__ import absolute_import, print_function, division
+from six.moves import range, zip
+
+from functools import reduce
+from itertools import chain
+
+import numpy
+
+from FIAT.reference_element import TensorProductCell
+
+import gem
+
+from finat.finiteelementbase import FiniteElementBase
+from finat.point_set import PointSet, TensorPointSet
+
+
+class TensorProductElement(FiniteElementBase):
+
+    def __init__(self, factors):
+        self.factors = tuple(factors)
+        assert all(fe.value_shape == () for fe in self.factors)
+
+        self._cell = TensorProductCell(*[fe.cell for fe in self.factors])
+        # self._degree = sum(fe.degree for fe in factors)  # correct?
+        # self._degree = max(fe.degree for fe in factors)  # FIAT
```

Comment:
For now, I just want to be clear that it isn't used yet, and listed the existing conventions for a future cleanup.